File: app/src/main/java/com/yuki/yukihub/util/TimeFormatUtil.java
package com.yuki.yukihub.util;

import java.text.DecimalFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class TimeFormatUtil {
    public static String playTime(long millis) {
        if (millis <= 0) return "0s";
        if (millis < 60000L) {
            long seconds = Math.max(1L, Math.round(millis / 1000.0));
            return seconds + "s";
        }
        double hours = millis / 3600000.0;
        if (hours < 1.0) {
            long minutes = Math.max(1L, Math.round(millis / 60000.0));
            return minutes + "m";
        }
        return new DecimalFormat("0.0").format(hours) + "h";
    }

    public static String date(long time) {
        if (time <= 0) return "从未游玩";
        return new SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.getDefault()).format(new Date(time));
    }
}

File: app/src/main/java/com/yuki/yukihub/util/TimeFormatUtil.kt
package com.yuki.yukihub.util

import java.text.DecimalFormat
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import kotlin.math.roundToLong

/**
 * 播放时长与日期格式化工具。
 */
object TimeFormatUtil {
    @JvmStatic
    fun playTime(millis: Long): String {
        if (millis <= 0) return "0s"
        if (millis < 60_000L) {
            val seconds = maxOf(1L, (millis / 1000.0).roundToLong())
            return "${seconds}s"
        }
        val hours = millis / 3_600_000.0
        if (hours < 1.0) {
            val minutes = maxOf(1L, (millis / 60_000.0).roundToLong())
            return "${minutes}m"
        }
        return DecimalFormat("0.0").format(hours) + "h"
    }

    @JvmStatic
    fun date(time: Long): String {
        if (time <= 0) return "从未游玩"
        return SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.getDefault()).format(Date(time))
    }
}
